spec/lib/citation_formatter_spec.rb

```diff
@@ -53,4 +53,101 @@
   it "sanitizes the MLA author name correctly in default citation if it's properly formatted" do
     expect(cit_gen.default_citations[mla.to_sym]).to include('Creator, Sample Parent')
   end
+
+  context '#abnormal_chars?' do
+    it 'returns true if abnormal character is found' do
+      ab_doc = document.dup.merge(creator_tesim: ["/"])
+      ab_doc2 = document.dup.merge(creator_tesim: ["1"])
+      ab_doc3 = document.dup.merge(creator_tesim: [","])
+      cit_gennie = described_class.new(ab_doc)
+      cit_gennie2 = described_class.new(ab_doc2)
+      cit_gennie3 = described_class.new(ab_doc3)
+
+      expect(cit_gennie.send(:abnormal_chars?)).to be_truthy
+      expect(cit_gennie2.send(:abnormal_chars?)).to be_truthy
+      expect(cit_gennie3.send(:abnormal_chars?)).to be_truthy
+    end
+
+    it 'returns false if abnormal character is not found' do
+      ab_doc = document.dup.merge(creator_tesim: ["ç"])
+      cit_gennie = described_class.new(ab_doc)
+
+      expect(cit_gen.send(:abnormal_chars?)).to be_falsey
+      expect(cit_gennie.send(:abnormal_chars?)).to be_falsey
+    end
+  end
+
+  context '#key_value_chunk_1' do
+    it "has the right keys" do
+      expect(cit_gen.send(:key_value_chunk_1).keys).to(
+        eq([:id, :abstract, :archive_location, :author, :"call-number", :edition, :institution])
+      )
+    end
+
+    it 'has the right values' do
+      keys_values_arr = [
```

Comment:
Is there a way to test the key-value chunks without converting them into arrays?

Reply:
Will take care of this in next push.